api/src/main/java/jakarta/data/Restrict.java

```diff
@@ -0,0 +1,232 @@
+        return new TextRestrictionRecord<>(field, Operator.EQUAL, value);
+    }
+
+    public static <T> Restriction<T> greaterThan(Comparable<Object> value, String field) {
+        return new BasicRestrictionRecord<>(field, Operator.GREATER_THAN, value);
+    }
+
+    public static <T> TextRestriction<T> greaterThan(String value, String field) {
+        return new TextRestrictionRecord<>(field, Operator.GREATER_THAN, value);
+    }
+
+    public static <T> Restriction<T> greaterThanEqual(Comparable<Object> value, String field) {
+        return new BasicRestrictionRecord<>(field, Operator.GREATER_THAN_EQUAL, value);
+    }
+
+    public static <T> TextRestriction<T> greaterThanEqual(String value, String field) {
+        return new TextRestrictionRecord<>(field, Operator.GREATER_THAN_EQUAL, value);
+    }
+
+    public static <T> Restriction<T> in(Set<Object> values, String field) {
+        return new BasicRestrictionRecord<>(field, Operator.IN, values);
+    }
+
+    public static <T> Restriction<T> lessThan(Comparable<Object> value, String field) {
+        return new BasicRestrictionRecord<>(field, Operator.LESS_THAN, value);
+    }
+
+    public static <T> TextRestriction<T> lessThan(String value, String field) {
+        return new TextRestrictionRecord<>(field, Operator.LESS_THAN, value);
+    }
+
+    public static <T> Restriction<T> lessThanEqual(Comparable<Object> value, String field) {
+        return new BasicRestrictionRecord<>(field, Operator.LESS_THAN_EQUAL, value);
+    }
+
+    public static <T> TextRestriction<T> lessThanEqual(String value, String field) {
+        return new TextRestrictionRecord<>(field, Operator.LESS_THAN_EQUAL, value);
+    }
+
+    // TODO this would be possible if Pattern is added, but is it even useful?
+    //public static <T> TextRestriction<T> like(Pattern pattern, String field) {
+    //    return new TextRestriction<>(field, Operator.LIKE, ESCAPED, pattern);
+    //}
+
+    public static <T> TextRestriction<T> like(String pattern, String field) {
+        return new TextRestrictionRecord<>(field, Operator.LIKE, pattern);
+    }
+
+    public static <T> TextRestriction<T> like(String pattern,
+                                               char charWildcard,
+                                               char stringWildcard,
+                                               String field) {
+        String p = toLikeEscaped(charWildcard, stringWildcard, false, pattern, false);
+        return new TextRestrictionRecord<>(field, Operator.LIKE, ESCAPED, p);
+    }
+
+    public static <T> Restriction<T> not(Object value, String field) {
+        return new BasicRestrictionRecord<>(field, NOT, Operator.EQUAL, value);
+    }
+
+    public static <T> TextRestriction<T> not(String value, String field) {
+        return new TextRestrictionRecord<>(field, NOT, Operator.EQUAL, value);
+    }
+
+    public static <T> TextRestriction<T> notContains(String substring, String field) {
```

Comment:
Can we remove all of those `no` method factory sequences?

It would be great if we have only two options:  by the method instead `negate`and a single method factory `not`.

It is pretty similar to the Java `Predicate`: https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/function/Predicate.html

Reply:
@otaviojava JDQL has operators like `not between`, `not like`, etc, so it makes sense to me for such things to be reflected here. And it's significantly less verbose.

Reply:
I agree with Gavin on this.

Being able to write code such as,

```
products.search(_Product.description.notContains("refurbished"));
```

is much more straightforward and readable than

```
products.search(_Product.description.contains("refurbished").negated());
```

or 

```
products.search(Restrict.not(_Product.description.contains("refurbished")));
```